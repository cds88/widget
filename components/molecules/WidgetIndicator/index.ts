export { default } from "./WidgetIndicator";
export { WIDGET_INDICATOR_DATA_TEST_IDS } from "./const.test";
