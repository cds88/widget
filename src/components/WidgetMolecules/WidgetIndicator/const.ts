export const WIDGET_INDICATOR_LEVELS = [20, 40, 60, 80, 100] as const