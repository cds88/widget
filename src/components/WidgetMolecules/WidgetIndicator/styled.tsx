import { Box } from '@mui/material';
import styled from 'styled-components';

export const IndicatorWrapperBox = styled(Box)`
  display: flex;
  flex-direction: row;
  gap: 1;
  flex: 1;
  justify-content: center;
  align-content: center;
`;

export const StyledIndicatorStripe = styled(Box)<{ $highlighted: boolean }>`
  flex: 1; // Set a width for the lines
  justify-self: center;
  height: 8px; // Height of the lines
  background-color: ${({ $highlighted }) =>
    $highlighted ? 'rgba(153,217,234)' : 'rgba(8,76,148)'};
  transition: background-color 0.3s;
  align-self: center;
  border-radius: 4px;
  margin-right: 4px;
`;
