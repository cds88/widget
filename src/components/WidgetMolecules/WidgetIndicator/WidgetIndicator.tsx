import { BoxProps } from '@mui/material';
import React from 'react';
import { IndicatorWrapperBox, StyledIndicatorStripe } from './styled';
import { WIDGET_INDICATOR_DATA_TEST_IDS } from './const.test';
import { WIDGET_INDICATOR_LEVELS } from './const';

type IndicatorProps = {
  percentage: number;
} & BoxProps;

const WidgetIndicator: React.FC<IndicatorProps> = ({
  percentage,
  ...props
}) => {
  return (
    <IndicatorWrapperBox
      data-testid={WIDGET_INDICATOR_DATA_TEST_IDS.WIDGET_INDICATOR_WRAPPER_BOX}
      {...props}
    >
      {WIDGET_INDICATOR_LEVELS.map((level, index) => {
        const highlighted = percentage >= level;
        return (
          <StyledIndicatorStripe
            data-testid={`${WIDGET_INDICATOR_DATA_TEST_IDS.WIDGET_INDICATOR_STRIPE}${index}`}
            aria-selected={highlighted? true: false}
            key={index}
            $highlighted={highlighted}
          />
        );
      })}
    </IndicatorWrapperBox>
  );
};

export default WidgetIndicator;
