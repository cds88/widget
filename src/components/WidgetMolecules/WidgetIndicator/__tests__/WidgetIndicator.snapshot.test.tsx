import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';
import WidgetIndicator from '../WidgetIndicator';

describe('WidgetIndicator', () => {
  test('should render the WidgetIndicator and match snapshot for 0% percentage', () => {
    const { asFragment } = render(<WidgetIndicator percentage={0} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should render the WidgetIndicator and match snapshot for 50% percentage', () => {
    const { asFragment } = render(<WidgetIndicator percentage={50} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should render the WidgetIndicator and match snapshot for 100% percentage', () => {
    const { asFragment } = render(<WidgetIndicator percentage={100} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
