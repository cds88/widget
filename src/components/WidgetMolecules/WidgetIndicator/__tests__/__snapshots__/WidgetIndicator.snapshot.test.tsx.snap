// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetIndicator should render the WidgetIndicator and match snapshot for 0% percentage 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb kyBNWc MuiBox-root css-0"
    data-testid="WIDGET_INDICATOR_WRAPPER_BOX"
  >
    <div
      aria-selected="false"
      class="sc-gtLWhw fXVKSa MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE0"
    />
    <div
      aria-selected="false"
      class="sc-gtLWhw fXVKSa MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE1"
    />
    <div
      aria-selected="false"
      class="sc-gtLWhw fXVKSa MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE2"
    />
    <div
      aria-selected="false"
      class="sc-gtLWhw fXVKSa MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE3"
    />
    <div
      aria-selected="false"
      class="sc-gtLWhw fXVKSa MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE4"
    />
  </div>
</DocumentFragment>
`;

exports[`WidgetIndicator should render the WidgetIndicator and match snapshot for 50% percentage 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb kyBNWc MuiBox-root css-0"
    data-testid="WIDGET_INDICATOR_WRAPPER_BOX"
  >
    <div
      aria-selected="true"
      class="sc-gtLWhw chVlSI MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE0"
    />
    <div
      aria-selected="true"
      class="sc-gtLWhw chVlSI MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE1"
    />
    <div
      aria-selected="false"
      class="sc-gtLWhw fXVKSa MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE2"
    />
    <div
      aria-selected="false"
      class="sc-gtLWhw fXVKSa MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE3"
    />
    <div
      aria-selected="false"
      class="sc-gtLWhw fXVKSa MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE4"
    />
  </div>
</DocumentFragment>
`;

exports[`WidgetIndicator should render the WidgetIndicator and match snapshot for 100% percentage 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb kyBNWc MuiBox-root css-0"
    data-testid="WIDGET_INDICATOR_WRAPPER_BOX"
  >
    <div
      aria-selected="true"
      class="sc-gtLWhw chVlSI MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE0"
    />
    <div
      aria-selected="true"
      class="sc-gtLWhw chVlSI MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE1"
    />
    <div
      aria-selected="true"
      class="sc-gtLWhw chVlSI MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE2"
    />
    <div
      aria-selected="true"
      class="sc-gtLWhw chVlSI MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE3"
    />
    <div
      aria-selected="true"
      class="sc-gtLWhw chVlSI MuiBox-root css-0"
      data-testid="WIDGET_INDICATOR_STRIPE4"
    />
  </div>
</DocumentFragment>
`;
